### DISCUSSION TITLE
Go Off Grid

### DISCUSSION THREAD

**Discussion by Barrstard** [TypQxQ/Sigenergy-Home-Assistant-Integration#30] [Q&A] [answered]
Thanks for this work! Really apprecuate it

Any luck with replicating the ‘Go Off Grid’ mode you can use in the app through modbus?
Trying to stop the system sending excess solar that is not used in the house to the grid when feed in tarifs are in the negative and the battery is full?

I can see registers for limiting how much solar is produced but I want to it react to how much energy the house is consuming. Just thought there might be something simple to replicate the ‘Go Off Grid’ mode

**Comment by TypQxQ** [comment, marked as answer]
Hi there!

This integration is towards the Modbus protocol as published by Sigenergy. I haven't seen anything like this yet. If they release it, I will implement it. 
You could do an automation that changes the maximum produced solar to be just under the consumption. Another automation to activate the former when prices are negative and battery is > 99% and disable when battery is <95%.

Hope this helps.

**Reply by Barrstard** [reply]
Brilliant! Thank you
And thanks again for the work you have done on this
